docs/open_models.md

```diff
@@ -1,24 +1,118 @@
+
+- `linux`: `CMAKE_ARGS="-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS"`
+- `macos` with Metal support: `CMAKE_ARGS="-DLLAMA_METAL=on"`
+- `windows`: `$env:CMAKE_ARGS = "-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS"`
+
+This will enable the `pip` installer to compile the `llama.cpp` with the proper hardware acceleration backend.
+
+Then run:
+
+```bash
+pip install llama-cpp-python
+```
+
+For our use case we also need to set up the web server that `llama-cpp-python` library provides. To install:
+
+```bash
+pip install 'llama-cpp-python[server]'
+```
+
+For detailed use consult the [`llama-cpp-python` docs](https://llama-cpp-python.readthedocs.io/en/latest/server/).
+
+Before we proceed we need to obtain the model weights in the `gguf` format. That should be a single file on your disk.
+
+In case you have weights in other formats check the `llama-cpp-python` docs for conversion to `gguf` format.
+
+Models in other formats `ggml`, `.safetensors`, etc. won't work without prior conversion to `gguf` file format with the solution described below!
+
+Which open model to use?
+==================
+
+Your best choice would be:
+
+- CodeLlama 70B
+- Mixtral 8x7B
+
+We are still testing this part, but the larger the model you can run the better. Sure the responses might be slower in terms of (token/s), but code quality will be higher.
+
+For testing that the open LLM `gpte` setup works we recommend starting with a smaller model. You can download weights of [CodeLlama-13B-GGUF by the `TheBloke`](https://huggingface.co/TheBloke/CodeLlama-13B-GGUF) choose the largest model version you can run (for example `Q6_K`), since quantisation will degrade LLM performance.
+
+Feel free to try out larger models on your hardware and see what happens.
 
 Running the Example
--------------------
+==================
+
+To see that your setup works check [test open LLM setup](examples/test_open_llm/README.md).
+
+If above tests work proceed 😉
+
+For checking that `gpte` works with the `CodeLLama` we recommend for you to create a project with `prompt` file content:
+
+```
+Write a python script that sums up two numbers. Provide only the `sum_two_numbers` function and nothing else.
+
+Provide two tests:
 
-Once the API is set up, you can find the host and the exposed TCP port by checking your Runpod dashboard.
+assert(sum_two_numbers(100, 10) == 110)
+assert(sum_two_numbers(10.1, 10) == 20.1)
+```
 
-Then, you can use the port and host to run the following example using WizardCoder-Python-34B hosted on Runpod:
+Now run the LLM in separate terminal:
 
+```bash
+python -m llama_cpp.server --model $model_path --n_batch 256 --n_gpu_layers 30
 ```
-  OPENAI_API_BASE=http://<host>:<port>/v1 python -m gpt_engineer.cli.main benchmark/pomodoro_timer --steps benchmark TheBloke_WizardCoder-Python-34B-V1.0-GPTQ
+
+Then in another terminal window set the following environment variables:
+
+```bash
+export OPENAI_API_BASE="http://localhost:8000/v1"
+export OPENAI_API_KEY="sk-xxx"
```

Comment:
Is this required? Can it not just null? Seems like an arbrirary magic string

Reply:
According to this: https://llama-cpp-python.readthedocs.io/en/latest/server/#multimodal-models yes. 

Although I'm not sure why. Since we use `llama-cpp-python` for open model inference I decided to follow their documentation.